src/instructlab/config.py

```diff
@@ -23,7 +23,7 @@
 DEFAULT_NUM_INSTRUCTIONS = 100
 DEFAULT_PROMPT_FILE = "prompt.txt"
 DEFAULT_GENERATED_FILES_OUTPUT_DIR = "generated"
-DEFAULT_CONNECTION_TIMEOUT = httpx.Timeout(timeout=30.0)
+DEFAULT_CONNECTION_TIMEOUT = httpx.Timeout(timeout=300.0)
```

Comment:
We need to think about this, waiting up to 5min seems too long. If the server crashes the chat prompt will hang for 5min, this degrades the user experience.
How about tweaking the value for the CPU use case only?

Reply:
There's little point to a lower timeout. If the server crashes, you'll get a broken pipe. If it doesn't crash and the timeout is too low, you'll get a weird network error while the server chugs along in the background. If you try it again, you'll be even more likely to get an error, since it didn't actually stop processing the previous request.

The only downside here is if your machine is too slow even for five minutes, you'll wait even longer before getting that error.

The short timeout is a real usability problem.

Reply:
This argument is true when running `ilab serve` + `ilab chat`, however when simply running `ilab chat`, a temporary server runs and if it crashes you would get the broken pipe on an incoming request, you just wait to 5 minutes.